maintenance-calendar/modules/ecs_scans/lambda/src/ignore.ts

```diff
@@ -0,0 +1,148 @@
+import assert from "assert";
+import pino from "pino";
+
+const logger = pino({ level: process.env.LOG_LEVEL ?? "debug" });
+const ssmClient = new SSMClient();
+
+type IgnoreSpec = {
+  name: string;
+  packageName: string;
+  packageVersion?: Array<string>;
+};
+
+type SnoozeList = {
+  cluster: string;
+  date: string;
+};
+
+assert(process.env.IGNORE_SPECS, "Ignore spec ARN is required");
+assert(process.env.SNOOZED_CLUSTERS, "Snoozed cluster ARN is required");
+const ignoreSpecARN = process.env.IGNORE_SPECS;
+const snoozedClustersARN = process.env.SNOOZED_CLUSTERS;
+
+const ignoreSpecs = async (arn: string): Promise<IgnoreSpec[]> => {
+  const getParameterResult = await ssmClient.send(
+    new GetParameterCommand({
+      Name: arn,
+      WithDecryption: true,
+    }),
+  );
+
+  assert(
+    getParameterResult?.Parameter?.Value,
+    "Failed to fetch ignore list from Parameter Store",
+  );
+  return JSON.parse(getParameterResult.Parameter.Value) as Array<IgnoreSpec>;
+};
+
+function attributeMatch(
+  attrs: Array<Attribute>,
+  key: string,
+  value: Exclude<IgnoreSpec[keyof IgnoreSpec], undefined>,
+): boolean {
+  for (const attr of attrs) {
+    if (attr.key === key) {
+      return Array.isArray(value)
+        ? value.some((v) => v === attr.value)
+        : value === attr.value;
+    }
+  }
+  return false;
+}
+
+function isFindingIgnoredBySpec(
+  finding: ImageScanFinding,
+  spec: IgnoreSpec,
+): boolean {
+  if (finding?.name !== spec.name) {
+    return false;
+  }
+
+  if (!finding.attributes) {
+    return false;
+  }
+
+  if (!attributeMatch(finding.attributes, "package_name", spec.packageName)) {
+    return false;
+  }
+
+  if (
+    spec.packageVersion &&
+    !attributeMatch(finding.attributes, "package_version", spec.packageVersion)
+  ) {
+    return false;
+  }
+
+  return true;
+}
+
+function isFindingIgnoredBySpecs(
+  finding: ImageScanFinding,
+  specs: Array<IgnoreSpec>,
+): boolean {
+  for (const spec of specs) {
+    if (isFindingIgnoredBySpec(finding, spec)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+const isFindingIgnored = async (
+  finding: ImageScanFinding,
+): Promise<boolean> => {
+  const ignored = await ignoreSpecs(ignoreSpecARN);
+  if (isFindingIgnoredBySpecs(finding, ignored)) {
+    return true;
+  }
+  return false;
+};
+
+const snoozeList = async (parameterARN: string): Promise<SnoozeList[]> => {
+  const getParameterResult = await ssmClient.send(
+    new GetParameterCommand({
+      Name: parameterARN,
+      WithDecryption: true,
+    }),
+  );
+
+  assert(
+    getParameterResult?.Parameter?.Value,
+    "Failed to fetch snooze list from Parameter Store",
+  );
+  return JSON.parse(getParameterResult.Parameter.Value) as Array<SnoozeList>;
+};
+
+const clusterSnoozed = async (
+  cluster: string,
+  snoozed: Array<SnoozeList>,
+): Promise<boolean> => {
+  const expiry = Date.now() - 604800000;
+  for (const snooze of snoozed) {
+    const snoozeDate = Date.parse(snooze.date);
```

Comment:
May seem like overkill but timezones are the literal worst; maybe we could use something like [js-joda](https://js-joda.github.io/js-joda/manual/ZonedDateTime.html#create-a-zoneddatetime) to do this datetime math? My thinking is that you'd parse `snooze.date` as midnight UTC by default and then compare `Date.now() - 7 days` to that. Otherwise we're at the mercy at whatever `Date.parse` decides the default TZ is.